### PR TITLE
core: add support for capella blocks

### PR DESCRIPTION
**Pull request by xenowits**
Adds support for capella blocks.

category: feature
ticket: #1464

### CODE REVIEW

File: core/fetcher/fetcher.go [hunk 1]
```diff
@@ -258,13 +258,18 @@ func (f *Fetcher) fetchProposerData(ctx context.Context, slot int64, defSet core
...
 
-		// Ensure fee recipient is correctly populated in block.
-		if block.Version == spec.DataVersionBellatrix {
-			actual := fmt.Sprintf("%#x", block.Bellatrix.Body.ExecutionPayload.FeeRecipient)
-			if !strings.EqualFold(actual, f.feeRecipientAddress) {
-				log.Warn(ctx, "Proposing block with unexpected fee recipient address", nil,
-					z.Str("expected", f.feeRecipientAddress), z.Str("actual", actual))
-			}
+		// Ensure fee recipient is correctly populated in block (if ever populated).
+		var actualAddr string
```
Comment (corverroos) [inline core/fetcher/fetcher.go:262]: extract this as a function

File: core/validatorapi/router.go [hunk 2]
```diff
@@ -562,8 +572,19 @@ func proposeBlindedBlock(p eth2client.BlindedBeaconBlockProposalProvider) handle
 
 func submitBlock(p eth2client.BeaconBlockSubmitter) handlerFunc {
 	return func(ctx context.Context, _ map[string]string, _ url.Values, body []byte) (interface{}, error) {
+		capellaBlock := new(capella.SignedBeaconBlock)
```
Comment (corverroos) [inline core/validatorapi/router.go:575]: please make sure we have a test for this, ie, when submitting a bellatrix block it gets marshalled as a bellatrix block, etc.

File: testutil/beaconmock/options.go [hunk 3]
```diff
@@ -453,7 +453,7 @@ func defaultMock(httpMock HTTPMock, httpServer *http.Server, clock clockwork.Clo
 							SyncCommitteeBits:      bitfield.NewBitvector512(),
 							SyncCommitteeSignature: testutil.RandomEth2Signature(),
 						},
-						ExecutionPayload: testutil.RandomExecutionPayLoad(),
+						ExecutionPayload: testutil.RandomBellatrixExecutionPayLoad(),
```
Comment (corverroos) [inline testutil/beaconmock/options.go:456]: lets return capella blocks from beaconmock be default.

File: core/validatorapi/router.go (hunk 2 above)
Comment (xenowits) [reply core/validatorapi/router.go:575]: I have added a test in `sigagg_test.go`(https://github.com/ObolNetwork/charon/blob/a8c623751c9bb5fa1330b6f90673e350ff71f686/core/sigagg/sigagg_test.go#L262). This tests marshalling & unmarshalling as it calls `block.clone()` which involves both steps.

Comment (corverroos) [reply core/validatorapi/router.go:575]: this code isn't tested by that package...

Comment (corverroos) [reply core/validatorapi/router.go:575]: we need to test the "submitBlock" function specifically

File: app/eth2wrap/synthproposer.go [hunk 4]
```diff
@@ -118,47 +120,11 @@ func (h *synthWrapper) BlindedBeaconBlockProposal(ctx context.Context, slot eth2
 	block, err := h.syntheticBlock(ctx, slot, vIdx)
 	if err != nil {
 		return nil, err
-	} else if block.Version != spec.DataVersionBellatrix {
+	} else if block.Version == spec.DataVersionPhase0 || block.Version == spec.DataVersionAltair {
```
Comment (corverroos) [inline app/eth2wrap/synthproposer.go:123]: rather handle this in a `default` case in switch case below

File: core/fetcher/fetcher.go [hunk 5]
```diff
@@ -379,3 +373,25 @@ func (f *Fetcher) fetchContributionData(ctx context.Context, slot int64, defSet
 
 	return resp, nil
 }
+
+// verifyFeeRecipient logs a warning when fee recipient is not correctly populated in the block.
+func verifyFeeRecipient(ctx context.Context, block *spec.VersionedBeaconBlock, feeRecipientAddress string) {
+	// Fee-recipient is not available in forks earlier than bellatrix.
+	if block.Version == spec.DataVersionPhase0 || block.Version == spec.DataVersionAltair {
```
Comment (corverroos) [inline core/fetcher/fetcher.go:380]: handle this default in switch case

**Review by corverroos — APPROVED** [review]

**Pull request merged**